torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -333,11 +333,34 @@ class ProcessGroupNCCL : public ProcessGroup {
       return !value_.isNone();
     }
 
+    std::shared_ptr<at::cuda::CUDAStream> getfutureNCCLCallbackStream() {
+      return futureNCCLCallbackStream_;
+    }
+
+    bool hasRecordStreamCallback() const {
+        return record_stream_cb_ != nullptr;
+    }
+
+    void setRecordStreamCallback(
+        std::function<void(const at::IValue&, at::cuda::CUDAStream)>
+            record_stream_cb) {
+      record_stream_cb_ = std::move(record_stream_cb);
+    }
+
+    void recordStream() {
+      TORCH_CHECK(record_stream_cb_);
```

Comment:
nit: I believe this should be `TORCH_INTERNAL_ASSERT` as it is a framework bug if this function is called with an invalid record_stream_cb_. 

I believe TORCH_CHECK is used for error validation coming from user input/code, and TORCH_INTERNAL_ASSERT for framework validation.

Reply:
Fixed.